homeassistant/components/tplink/coordinator.py

```diff
@@ -41,9 +54,24 @@ def __init__(
 
     async def _async_update_data(self) -> None:
         """Fetch all device and sensor data from api."""
+        device = self.device
         try:
-            await self.device.update(update_children=False)
+            await device.update(update_children=False)
         except AuthenticationException as ex:
             raise ConfigEntryAuthFailed from ex
         except SmartDeviceException as ex:
             raise UpdateFailed(ex) from ex
+        if self.has_emeter:
+            with contextlib.suppress(AttributeError):
+                # clear cached property on update
+                del self.emeter_realtime
+
+    @cached_property
+    def emeter_realtime(self) -> EmeterStatus | None:
```

Comment:
It would have been nice to do this in the lib but we can't use `@cached_property` there until python 3.12 is the min supported because of the undocumented locking behavior